### DISCUSSION TITLE
How do I get more debug information for "thread '<unknown>' has overflowed its stack"

### DISCUSSION THREAD

**Discussion by phillipuniverse** [astral-sh/ruff#5523] [Q&A] [answered]
In some of my projects with various Ruff releases (like 0.0.277 for instance) I upgrade and get a message like this:

```
~/someproject ‹dependabot/pip/ruff-0.0.277*› » poetry run ruff . --fix

thread '<unknown>' has overflowed its stack
fatal runtime error: stack overflow
zsh: abort      poetry run ruff . --fix
```

I'm not sure what to do in this scenario or how to start opening a bug. Any advice?

More information:

- I have seen this issue since 0.0.275
- Below is my `pyproject.toml` config. If I comment all of the `select` out, no issues. If I enable any rule _except for `I` (isort)_ then I everything works normally. If I leave any of the others enabled, I get the above error

```toml
[tool.ruff]
target-version = "py311"
line-length = 100

select = [
    "I",  # isort
    "C4", # flake8-comprehensions
    "UP", # pyupgrade
    "F",  # pyflakes
    "E",  # pycodestyle
]
ignore = [
    "E501",  # line too long, handled by black
    "C401",  # generator syntax for sets vs always force set comprehension
    "PLC0414",  # allow explicit re-exports using 'as' without forcing __all__
]

    [tool.ruff.isort]
    known-first-party = ["app", "tests"]
```

**Comment by charliermarsh** [comment, marked as answer]
There must be some kind of instability in the import sorter, something that's causing an infinite loop. You could run with `RAYON_NUM_THREADS=1 ruff check --verbose --no-cache .`, which should list out each file as it checks it. Use that to determine which file is erroring, and see if you can trim the file down to a minimal reproduction -- that'd be really useful.

**Reply by phillipuniverse** [reply]
@charliermarsh super helpful, this did indeed help me find the offending files.

Now that I have it somewhat narrowed down I can dig in more. The 2 problematic files are ~3,100 and ~4,600 lines each, I'll see if I can locate some problematic areas.

Thanks!
